handbook/engineering/continuous_releasability.md

```diff
@@ -0,0 +1,64 @@
+# Continuous releasability
+
+We practice **continuous releasability**. This is a variant of [continous
+delivery](https://en.wikipedia.org/wiki/Continuous_delivery) adapted to a monthly release cadence
+that serves the needs of customers running their own Sourcegraph instances.
+
+## Continuous Releasability Contract
+
+* `master` branch is always releasable
+* Every new feature includes a feature flag
+
+### `master` is always releasable
+
+Every change merged into `master` must preserve the releasability of `master`. *Releasable* means
+that the revision can be released to customers. In particular,
+
+* **Up-to-date changelog and docs**: The changelog and documentation reflect the current feature set.
+  * Docs must be updated as soon as a feature is merged and enabled.
+  * You can merge a feature without yet updating the docs if it is behind a disabled feature flag.
+* **QA**: Every feature has been QA'd by an appropriate set of users in an appropriate environment.
+  * Simple bug fixes should be tested at the very least by the author.
+  * Simple features should be tested by someone other than the author.
+  * Complex or scale-sensitive changes should be tested by multiple people on one of the dogfood instances.
```

Comment:
This was an idea I discussed with @sqs - the author would demo their change to product to validate documentation for that feature as well as usability. My concern is with becoming a bottleneck, though my expectation is that this would be a quick process assuming there aren't any issues with the change. Curious to hear thoughts/concerns/other ideas.

```suggestion
  * Complex or scale-sensitive changes should be tested by multiple people on one of the dogfood instances.
  * Changes that impact the user and new product features should be demoed with Product to verify workflows, documentation, and usability.
```